tutorials/interactive/index.md

```diff
@@ -0,0 +1,814 @@
+  from GitHub.
+* Create an `images` directory in your project, and add
+  [`lakes.jpg`](https://github.com/flutter/website/blob/master/_includes/_code/lakes/images/lake.jpg)
+  from GitHub.
+
+Once you have a connected and enabled device, or you've launched the
+[iOS simulator](/setup/#set-up-the-ios-simulator) (part of the Flutter install),
+you are good to go!
+</aside>
+
+
+In [Building Layouts for Flutter](https://flutter.io/tutorials/layout/),
+we showed how to create the layout for the followng screenshot.
+
+<img src="images/lakes.png" style="border:1px solid black" alt="The starting Lakes app that we will modify">
+
+When the app first launches, the star is solid red, indicating that this lake
+has previously been favorited. The number next to the star indicates that a
+total of 41 people have favorited this lake.  After completing this tutorial,
+tapping the star removes its favorited status, replacing
+the solid star with an outline, and decreasing the count. Tapping
+again favorites the lake, drawing a solid star and increasing the count.
+
+The following shows both states.
+
+<img src="images/favorited-not-favorited.png" alt="the custom widget you'll create">
+
+To accomplish this, you'll create a single custom widget that includes both the
+star and and the count. Tapping the star changes state for both widgets,
+so the same widget should manage both.
+
+If you don't already have the starting code for the Lakes example,
+get it from the links in the blue [**Note**](#note) box.
+
+### Contents
+
+* [Stateful and stateless widgets](#stateful-stateless)
+* [Creating a stateful widget](#creating-stateful-widget)
+  * [Step 1: Decide who manages the widget's state](#step-1)
+  * [Step 2: Subclass StatefulWidget](#step-2)
+  * [Step 3: Subclass the State object](#step-3)
+  * [Step 4: Plug the stateful widget into the widget tree](#step-4)
+  * [Problems?](#problems)
+* [Managing state](#managing-state)
+  * [The widget manages its own state](#self-managed)
+  * [The parent manages the widget's state](#parent-managed)
+  * [A mix-and-match approach](#mix-and-match)
+* [Other interactive widgets](#other-interactive-widgets)
+  * [Standard widgets](#standard-widgets)
+  * [Material widgets](#material-widgets)
+* [Resources](#resources)
+
+<a name="stateful-stateless"></a>
+## Stateful and stateless widgets
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* All widgets are either stateful or stateless.
+* _State_ means information about a widget that might change.
+  A checkbox's state, for example, would probably
+  include a boolean indicating whether it's currently checked.
+* If a widget changes based on user interaction, or as the result of a
+  data feed (for example), it's _stateful_.
+* All widgets should be _immutable_, meaning once created they cannot change.
```

Comment:
But in the previous bullet point you just said that a widget could change.

I think this is confusing "class that inherits from Widget" with "user interaction control", both of which could be called "widget".